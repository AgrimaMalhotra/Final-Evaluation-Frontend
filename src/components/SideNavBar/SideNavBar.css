.sidebar {
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  width: 15%;
  height: 100vh;
}

.header {
  display: flex;
  align-items: center;
  background-color:#7136FA;
  height: 45px;
  color: white;
  font-size: 1.8rem;
  font-weight: bolder;
  padding-left: 20px;
}

.collections {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  background-color: rgb(49, 44, 44);
  padding-top: 30px;
}

.collections-header {
  font-size: 0.9rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-weight: bolder;
  color: whitesmoke;
  margin-bottom: 18px;
  padding-left: 20px;
  padding-right: 10px;
}

.collections-list {
  padding-left: 20px;
  color: rgb(134, 134, 134);
}

ul{  padding-left: 20px;}
.collections-list li {
  margin-bottom: 18px;
}

.collections-list li:active{
  background-color: #000;
  color: white;
  padding-left: 0;
}
.collections-list li:hover{
  background-color: #000;
  color: white;
  padding-left: 0;
}
.content-builder {
  color: white;
  font-size: 0.9rem;
  padding: 6px;
  font-weight: 600;
  margin-top: 15px;
  padding-left: 20px;
}

.content-builder:active, .content-builder:hover {
  background-color: #000;
    color: white;
}