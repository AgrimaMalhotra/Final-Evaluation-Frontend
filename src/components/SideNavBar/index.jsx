import * as React from 'react';
import './SideNavBar.css';
import searchIcon from '../../assets/icon-search-dark_2023-03-09/icon-search-dark.png';
import propTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import {  DASHBOARD_ROUTE } from '../../constants/routes';

const SideNavBar = ({collectionNames}) => { 
  const navigate =  useNavigate();

  return (
    <div className='sidebar'>
      <div className='header'>
        <p>CMS+</p>
      </div>
      <div className="collections-nav-bar">
        <div className="collections-header">
          <p>COLLECTION TYPES</p>
          <img src={searchIcon}/>
        </div>
        <div className='collections-list'>
          <ul>
            {collectionNames.map((collectionName) => (
              <li key={collectionName.id} onClick={() => 
                navigate(`/collections/${collectionName.id}`) }>{collectionName.name}</li>)
            )}
          </ul>
        </div>
        <div className="content-builder" onClick={() => 
          navigate(DASHBOARD_ROUTE)}>
          <p>CONTENT TYPE BUILDER</p></div>
      </div>
    </div>
  );
};

SideNavBar.propTypes = {
  collectionNames: propTypes.array.isRequired,
};

export default SideNavBar;