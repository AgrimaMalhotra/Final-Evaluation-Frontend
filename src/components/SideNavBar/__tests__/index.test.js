import React from 'react';
import {render} from '@testing-library/react';
import SideNavBar from '../index';

jest.mock('react-router-dom', () => ({ useNavigate: () => jest.fn() }));
describe('SideNavBar', () => {
  const mockCollectionNames = ['collection1','collection2','collection3'];
  it('should render correctly', () => {
    const component = render(<SideNavBar collectionNames={mockCollectionNames}/>);
    expect(component).toMatchSnapshot();
  });
});
