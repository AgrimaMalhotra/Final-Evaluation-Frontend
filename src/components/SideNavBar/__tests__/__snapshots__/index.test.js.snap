// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavBar should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sidebar"
      >
        <div
          class="header"
        >
          <p>
            CMS+
          </p>
        </div>
        <div
          class="collections-nav-bar"
        >
          <div
            class="collections-header"
          >
            <p>
              COLLECTION TYPES
            </p>
            <img
              src="icon-search-dark.png"
            />
          </div>
          <div
            class="collections-list"
          >
            <ul>
              <li />
              <li />
              <li />
            </ul>
          </div>
          <div
            class="content-builder"
          >
            <p>
              CONTENT TYPE BUILDER
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sidebar"
    >
      <div
        class="header"
      >
        <p>
          CMS+
        </p>
      </div>
      <div
        class="collections-nav-bar"
      >
        <div
          class="collections-header"
        >
          <p>
            COLLECTION TYPES
          </p>
          <img
            src="icon-search-dark.png"
          />
        </div>
        <div
          class="collections-list"
        >
          <ul>
            <li />
            <li />
            <li />
          </ul>
        </div>
        <div
          class="content-builder"
        >
          <p>
            CONTENT TYPE BUILDER
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
